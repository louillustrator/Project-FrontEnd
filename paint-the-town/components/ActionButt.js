import React from "react";
import { StyleSheet } from "react-native";
import ActionButton from "react-native-circular-action-menu";
import Icon from "react-native-vector-icons/Ionicons";

const ActionButt = ({
  colour,
  navigation,
  changeColour,
  stop,
  start,
  pause,
  setShowSlider
}) => {
  const colourPickerFn = () => {
    navigation.navigate("ColourPicker", {
      currentColour: colour,
      changeColour: changeColour
    });
  };
  const rgb = hexToRGB(colour);
  return (
    <ActionButton
      buttonColor={rgb}
      radius={90}
      position="center"
      degrees={270}
      icon={<Icon name="md-brush" style={styles.actionButtonIcon} />}
      outRangeScale={0.8}
      style={{ position: "absolute", top: 200, left: 50, zIndex: 10 }}
    >
      <ActionButton.Item
        buttonColor="#228B22"
        title="Start"
        size={60}
        onPress={start}
        style={{ zIndex: 10 }}
      >
        <Icon name="md-play" style={styles.actionButtonIcon} />
      </ActionButton.Item>
      <ActionButton.Item
        buttonColor="#3498db"
        title="Pause"
        size={60}
        onPress={pause}
        style={{ zIndex: 10 }}
      >
        <Icon name="md-pause" style={styles.actionButtonIcon} />
      </ActionButton.Item>
      <ActionButton.Item
        buttonColor="#E74C3C"
        title="Stop"
        size={60}
        onPress={stop}
        style={{ zIndex: 10 }}
      >
        <Icon name="md-square" style={styles.actionButtonIcon} />
      </ActionButton.Item>
      <ActionButton.Item
        buttonColor={colour || "#9b59b6"}
        title="Colour Select"
        size={60}
        onPress={colourPickerFn}
        style={{ zIndex: 10 }}
      >
        <Icon name="md-color-palette" style={styles.actionButtonIcon} />
      </ActionButton.Item>
      <ActionButton.Item 
        buttonColor='#9b59b6' 
        title="Width Select" 
        size={60} 
        onPress={() => setShowSlider(true)} 
        style={{zIndex: 10}}>
            <Icon name="md-arrow-back" style={styles.actionButtonIcon} />
            <Icon name="md-arrow-forward" style={styles.actionButtonIcon} />
      </ActionButton.Item>
    </ActionButton>
  );
};

const hexToRGB = (hex) => {
  const r = parseInt(hex.slice(1, 3), 16),
        g = parseInt(hex.slice(3, 5), 16),
        b = parseInt(hex.slice(5, 7), 16);
  return `rgba(${r},${g},${b},1)`;
}

const styles = StyleSheet.create({
  actionButtonIcon: {
    fontSize: 20,
    height: 22,
    color: "white",
    zIndex: 10
  }
});
export default ActionButt;
