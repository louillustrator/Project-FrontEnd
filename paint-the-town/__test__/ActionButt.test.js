import React from "react";
import { mount } from 'enzyme';
import renderer from "react-test-renderer";
import ActionButt from "../components/ActionButt";


describe('ActionButt', () => {
    let props;
    beforeEach(() => {
        props = {
            colour: "#228B22",
            navigation: undefined,
            changeColour: undefined,
            stop: undefined,
            start: undefined,
            pause: undefined,
            setShowSlider: undefined
        };
    });
    it('renders without crashing', () => {
        let wrapper = mount(<ActionButt {...props} />);
    });
    it("renders correctly", () => {
        const tree = renderer.create(<ActionButt {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});