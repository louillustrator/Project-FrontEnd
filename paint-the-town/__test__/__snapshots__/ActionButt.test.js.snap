// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButt renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "flexDirection": "column",
          "padding": 10,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          Object {
            "height": 63,
            "width": 63,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(34, 139, 34, 1)",
              "borderRadius": 31.5,
              "height": 63,
              "justifyContent": "center",
              "shadowColor": "#444",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 63,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
